pandas/core/generic.py

```diff
@@ -8786,6 +8787,9 @@ def _where(
         """
         inplace = validate_bool_kwarg(inplace, "inplace")
 
+        if axis is not None:
```

Comment:
we do this i am pretty sure in the caller, so why do you need it here? (e.g. when is this ever non-None)?

Reply:
for starters the default is None, and no axis is passed when this is called from `Series.__setitem__`